### PR TITLE
Resolve WCAG 2.1 AA issues

### PR DESCRIPTION
**Pull request by bjankord**
### Summary
Updating test structure for list items and adding aria roles for optgroups in the form-select based on https://github.com/w3c/aria/issues/719

This uses a version of terra-toolkit with the following rules enabled in axe-core:
```
'wcag2a', 'wcag2aa', 'wcag21aa', 'section508'
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -95,10 +95,10 @@
     "stylelint": "^10.0.1",
     "stylelint-config-terra": "^3.0.0",
     "terra-aggregate-translations": "^1.0.0",
-    "terra-dev-site": "^5.0.0",
+    "terra-dev-site": "cerner/terra-dev-site#update-axe-rules",
```
Comment (StephenEsser) [inline package.json]: What is this dark magic.

Comment (bjankord) [reply package.json]: https://medium.com/@jonchurch/use-github-branch-as-dependency-in-package-json-5eb609c81f1a

Comment (StephenEsser) [reply package.json]: giphy

**Comment by emilyrohrbough** [comment]
@bjankord will you be opening a pr in toolkit to add these?

**Comment by bjankord** [comment]
@emilyrohrbough Yeah I plan to update the axe-core rules to `'wcag2a', 'wcag2aa', 'wcag21aa', 'section508'` in terra-toolkit. Right now I've just been testing on these branches of [terra-toolkit](https://github.com/cerner/terra-toolkit/tree/update-axe-rules) and [terra-dev-site](https://github.com/cerner/terra-dev-site/tree/update-axe-rules) to resolve the issues before I turn on the rules in terra-core.

File: packages/terra-list/src/terra-dev-site/test/list/ListItem.test.jsx [hunk 2]
```diff
@@ -3,33 +3,37 @@ import React from 'react';
...
+        <p>test default</p>
+      </Item>
+      <Item
+        key="chevron"
+        hasChevron
+      >
+        <p>test chevron</p>
+      </Item>
+    </List>
+    <List role="listbox">
```
Comment (emilyrohrbough) [inline packages/terra-list/src/terra-dev-site/test/list/ListItem.test.jsx]: why do we need to add a different role here? Is this needed for selectable lists?

Comment (bjankord) [reply packages/terra-list/src/terra-dev-site/test/list/ListItem.test.jsx]: Correct, selectable lists need the listbox role added to the `<List />` component. [We note that in List readme.](https://github.com/cerner/terra-core/blob/b9d9846485813b17eb8fa581ea757c1445e2fba6/packages/terra-list/docs/List.md#terra-list)

**Pull request merged**